libkbfs/folder_branch_ops.go

```diff
@@ -1018,19 +1066,33 @@ func (fbo *folderBranchOps) getMDForReadNeedIdentify(
 // one must be created by the caller.
 func (fbo *folderBranchOps) getMDForReadNeedIdentifyOnMaybeFirstAccess(
 	ctx context.Context, lState *lockState) (ImmutableRootMetadata, error) {
-	irmd, err := fbo.getMDForReadHelper(ctx, lState, mdReadNeedIdentify)
+	md, err := fbo.getMDLockedForRead(ctx, lState, mdReadNeedIdentify)
 
 	if _, ok := err.(MDWriteNeededInRequest); ok {
 		fbo.mdWriterLock.Lock(lState)
 		defer fbo.mdWriterLock.Unlock(lState)
-		irmd, err = fbo.getMDForReadHelper(ctx, lState, mdWrite)
+		md, err = fbo.getMDLockedForWriteOrRekey(ctx, lState, false)
 	}
 
 	if _, noMD := errors.Cause(err).(NoMergedMDError); noMD {
 		return ImmutableRootMetadata{}, nil
 	}
 
-	return irmd, err
+	if err != nil {
+		return ImmutableRootMetadata{}, err
+	}
+
+	if !md.TlfID().IsPublic() {
```

Comment:
Previously this was going though `getMDForReadHelper` which includes the check. After separating the reads and write calls properly this is the only place where `mdWrite` was used together with `getMDForReadHelper` - so inline that check rather than creating a single use separate `getMDForWriteHelper`.